src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs

```diff
@@ -0,0 +1,773 @@
+                            _actionsList.Add(des.Conflicts());
+                            break;
+                        case "LegacyAttachments":
+                            _actionsList.Add(des.Documents());
+                            break;
+                        case "LegacyDocumentDeletions":
+                            _actionsList.Add(des.LegacyDocumentDeletions());
+                            break;
+                        case "LegacyAttachmentDeletions":
+                            _actionsList.Add(des.LegacyAttachmentDeletions());
+                            break;
+                        case "LegacyCounters":
+                            _actionsList.Add(des.LegacyCounters(_result));
+                            break;
+                        case "Counters":
+                            _actionsList.Add(des.Counters(_result));
+                            break;
+                        case "TimeSeries":
+                            _actionsList.Add(des.TimeSeries());
+                            break;
+                        case nameof(DatabaseItemType.CompareExchange):
+                            _actionsList.Add(des.CompareExchange(_context));
+                            break;
+                    }
+                }
+            }
+        }
+
+        private async Task SendBatch()
+        {
+            await CleanActionList();
+            foreach (var des in _desList)
+            {
+                await des.DisposeAsync();
+            }
+
+            var task = SendImportBatch();
+            task.Wait();
+            _itemsInBatch = 0;
+            _desList.Clear();
+        }
+
+        public async Task SendImportBatch()
+        {
+            var tasks = new List<Task>();
+            for (int i = 0; i < _shardedContext.ShardCount; i++)
+            {
+                _streamList[i].Position = 0;
+
+                var multi = new MultipartFormDataContent
+                {
+                    {
+                        new BlittableJsonContent(async stream2 => await _context.WriteAsync(stream2, _optionsAsBlittable).ConfigureAwait(false)),
+                        Constants.Smuggler.ImportOptions
+                    },
+                    {new Client.Documents.Smuggler.DatabaseSmuggler.StreamContentWithConfirmation(_streamList[i], _tcs), "file", "name"}
+                };
+                var cmd = new ShardedImportCommand(_handler, Headers.None, multi);
+
+                var task = _shardedContext.RequestExecutors[i].ExecuteAsync(cmd, _contextList[i]);
+
+                tasks.Add(task);
+            }
+
+            await tasks.WhenAll();
+            for (int i = 0; i < _shardedContext.ShardCount; i++)
+            {
+                _streamList[i].Position = 0;
+            }
+        }
+
+        private void DisposeSourceInfo()
+        {
+            if (_source is StreamSource streamSource)
+            {
+                foreach (var iDisposable in streamSource._toDispose)
```

Comment:
naming
And why are you accessing a private variable in `streamSource`, why not dispose it directly?